mypy/types.py

```diff
@@ -1341,10 +1340,7 @@ def __init__(self, item: Union[Instance, AnyType, TypeVarType, TupleType, NoneTy
         type UnionType must be handled through make_normalized static method.
         """
         super().__init__(line, column)
-        if isinstance(item, CallableType) and item.is_type_obj():
-            self.item = item.fallback
-        else:
-            self.item = item
+        self.item = item
```

Comment:
Why does this change help? Is it because the `type` attribute of an `Instance` might refer to `FakeInfo` during deserialization?

Reply:
Yes.

Reply:
(Or previously to ``None``)